### PR TITLE
Apollo portal playbook

### PR DESCRIPTION
**Pull request by neoformit**
Required the following in ansible/playbooks/group_vars/newportalvms/vault:

```
admin_email:           <str>
admin_password:        <str>
user_support_email:    <str>
postgres_db_password:  <str>
django_secret_key:     <str>
smtp_password:         <str>
slack_error_api_key:   <str>
recaptcha_public_key:  <str>
recaptcha_private_key: <str>
```

### CODE REVIEW

File: ansible/playbooks/group_vars/newportalvms/vars [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+# Admin user login for the admin portal
+admin_user:
+  first_name: admin
+  last_name: admin
+  email: "{{ admin_email }}"
+  password: "{{ admin_password }}"
+
+smtp:
+  hostname: smtp.office365.com
```
Comment (jlqfab) [inline ansible/playbooks/group_vars/newportalvms/vars:67]: This needs to use the `mail.usegalaxy.org.au` postal server as per the current Drupal apollo-portal

File: ansible/playbooks/hosts [hunk 2]
```diff
@@ -47,11 +47,11 @@ apollo-monitor.genome.edu.au
 apollo-portal.genome.edu.au
 #apollo-portal-main.genome.edu.au # use this when apollo-portal redirects to apollo-portal-offline
 apollo-deploy.genome.edu.au
-ubuntu22-test.genome.edu.au
 ubuntu20-test.genome.edu.au
 #ubuntu18-test.genome.edu.au
 
-
+[dev_portal_webservers]
+ubuntu22-test.genome.edu.au  ansible_connection=ssh  ansible_user=ubuntu
```
Comment (jlqfab) [inline ansible/playbooks/hosts:55]: this should go in it's own inventory file. See `ansible/playbooks/buildmonitor.inventory` for example.

File: ansible/playbooks/playbook-dev-deploy-apollo-portal.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+---
+- hosts: dev_portal_webservers
```
Comment (jlqfab) [inline ansible/playbooks/playbook-dev-deploy-apollo-portal.yml:2]: this should be `newportalvms` to match the group_vars.

File: ansible/playbooks/hosts (hunk 2 above)
Comment (jlqfab) [reply ansible/playbooks/hosts:55]: also it should be defined as `newportalvms` to match the group_vars.

File: ansible/playbooks/playbook-dev-deploy-apollo-portal.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  remote_user: ubuntu
+  become: yes
+  become_method: sudo
+  connection: "ssh"
+
+  vars_files:
```
Comment (jlqfab) [inline ansible/playbooks/playbook-dev-deploy-apollo-portal.yml:8]: not needed if define the VM as a member of `newportalvms` in the inventory

File: ansible/roles/geerlingguy.ansible/.github/workflows/release.yml
Comment (jlqfab) [inline ansible/roles/geerlingguy.ansible/.github/workflows/release.yml:1]: Is this applicable to this project or this this galaxy specific??? Same goes for the other `geerlinguy.ansible/.github` files - are they needed?

File: ansible/roles/geerlingguy.ansible/README.md
Comment (jlqfab) [inline ansible/roles/geerlingguy.ansible/README.md:1]: why is this here? the `geerlingguy.ansible` role is installed via the `ansible-galaxy/requirements.yml` on the deployment server. It is not required on wanted on the web server.

File: ansible/roles/geerlingguy.ansible/.ansible-lint
Comment (jlqfab) [inline ansible/roles/geerlingguy.ansible/.ansible-lint:1]: I think the geerlingguy.ansible role should be removed - it isn't (or shouldn't be) part of the portal deployment. The geerlingguy.ansible role is installed via the ansible-galaxy/requirements.yml on the deployment server. It is not required on wanted on the web server.

File: ansible/roles/geerlingguy.git/.ansible-lint
Comment (jlqfab) [inline ansible/roles/geerlingguy.git/.ansible-lint:1]: the geerlingguy.git role should be installed via ansible-galaxy if required, not included in the apollo infrastructure code base

File: ansible/roles/geerlingguy.postgresql/README.md
Comment (jlqfab) [inline ansible/roles/geerlingguy.postgresql/README.md:1]: As with the other geerlingguy ansible roles - these should be installed via ansible-galaxy, _not_ included in the apollo infrastructure code base.

**Review by jlqfab — CHANGES_REQUESTED** [review]
As it stands. This can't be merged. It includes geerlingguy roles into the codebase, when they only need to be added to the `ansible-galaxy/requirements.yml` file. Also, smtp settings need to be corrected (see comments and as per conversations), and a separate inventory file should be created that adds the VM to the `newportalvms` group, which in turn provides access to those group_vars.

File: ansible/playbooks/group_vars/newportalvms/vars (hunk 1 above)
Comment (neoformit) [reply ansible/playbooks/group_vars/newportalvms/vars:67]: Yeah I'm working on getting the Postal server to work - I can make a separate PR for this.

File: ansible/roles/geerlingguy.ansible/.github/workflows/release.yml
Comment (neoformit) [reply ansible/roles/geerlingguy.ansible/.github/workflows/release.yml:1]: I don't know - `galaxy-ansible` installed them. So we should either commit, or gitignore them?

File: ansible/roles/geerlingguy.ansible/README.md
Comment (neoformit) [reply ansible/roles/geerlingguy.ansible/README.md:1]: Then we should add this to the gitignore?

**Comment by neoformit** [comment]
I have reset the last commit (the portal role/playbook) and added the following to `.gitignore`:

```
# Development roles
ansible/roles/geerlingguy.ansible/
ansible/roles/geerlingguy.git/
```

Then committed the changes again. Let me know if you're happy with this and I will force-push (this will wipe those roles from the repo history). I have also fixed the inventory files and host name.

**Comment by jlqfab** [comment]
The  ansible-galaxy roles from geerlingguy should be installed in your `$HOME/.ansible/roles` directory when you install with ansible-galaxy. So they shouldn't be turning up under the code repo.

File: ansible/playbooks/hosts [hunk 5]
```diff
@@ -47,12 +47,9 @@ apollo-monitor.genome.edu.au
 apollo-portal.genome.edu.au
 #apollo-portal-main.genome.edu.au # use this when apollo-portal redirects to apollo-portal-offline
 apollo-deploy.genome.edu.au
-ubuntu22-test.genome.edu.au
```
Comment (jlqfab) [inline ansible/playbooks/hosts:50]: can you put ubuntu22-test.genome.edu.au back in the hosts inventory file. ie just revert ansible/playbooks/hosts  to what it was before you made any changes (ie it shouldn't be included in this PR). otherwise I need to add it back for system updates.

**Review by jlqfab — APPROVED** [review]
Thanks. Looks good!
